neqo-transport/src/connection/tests/stream.rs

```diff
@@ -539,6 +539,42 @@ fn do_not_accept_data_after_stop_sending() {
     );
 }
 
+#[test]
+/// Server sends a number of stream-related frames for a client-initiated stream that is not yet
+/// created. This should cause the client to close the connection.
+fn illegal_stream_frames() {
+    struct IllegalWriter(Vec<u64>);
+
+    impl crate::connection::test_internal::FrameWriter for IllegalWriter {
+        fn write_frames(&mut self, builder: &mut PacketBuilder) {
+            builder.write_varint_frame(&self.0);
+        }
+    }
+
+    fn test_with_illegal_frame(frame: &[u64]) {
+        let mut client = default_client();
+        let mut server = default_server();
+        connect(&mut client, &mut server);
+        let dgram = send_with_extra(&mut server, IllegalWriter(frame.to_vec()), now());
+        client.process_input(dgram, now());
+        assert!(client.state().closed());
+    }
+
+    // 0 = Client-Initiated, Bidirectional; 2 = Client-Initiated, Unidirectional
+    for stream_id in [0, 2] {
+        // Illegal RESET_STREAM frame
+        test_with_illegal_frame(&[FRAME_TYPE_RESET_STREAM, stream_id, 0, 0]);
+        // Illegal STOP_SENDING frame
+        test_with_illegal_frame(&[FRAME_TYPE_STOP_SENDING, stream_id, 0]);
+        // Illegal MAX_STREAM_DATA frame
+        test_with_illegal_frame(&[FRAME_TYPE_MAX_STREAM_DATA, stream_id, 0]);
+        // Illegal STREAM_DATA_BLOCKED frame
+        test_with_illegal_frame(&[FRAME_TYPE_STREAM_DATA_BLOCKED, stream_id, 0]);
+        // Illegal STREAM frame
+        test_with_illegal_frame(&[FRAME_TYPE_STREAM_CLIENT_INI_BIDI, stream_id, 0]);
```

Comment:
HAXX:

```suggestion
        for frame_type in [FRAME_TYPE_RESET_STREAM, ...] {
            test_with_illegal_frame(&[frame_type, stream_id, 0, 0]);
        }
```

Observe how only `RESET_STREAM` differs from the others.  But the extra 0 byte on the others will be treated as a `PADDING` frame.